src/ocl/convolutionocl.cpp

```diff
@@ -2036,51 +2064,67 @@ void ConvolutionDescriptor::FindConvBwdDataAlgorithm(Handle& handle,
-                EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
-            }
-
-            // Direct algo
-            if(!use_winograd_only)
-            {
-                ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
-                bufs.SetBwd(dx, w, dy);
-                const auto all = FindDataDirectSolutions(
-                    handle, dxDesc, wDesc, dyDesc, exhaustiveSearch, false, bufs, invoke_ctx);
-                const auto algorithm_name = AlgorithmName{"miopenConvolutionBwdDataAlgoDirect"};
-                PrecompileSolutions(handle, all);
-                EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
+                if(!use_winograd_only)
+                {
+                    ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
+                    bufs.SetBwd(dx, w, dy);
+                    const auto direct_solutions = FindDataDirectSolutions(
+                        handle, dxDesc, wDesc, dyDesc, exhaustiveSearch, false, bufs, invoke_ctx);
+                    const auto implictgemm_solutions = this->FindDataImplicitGemmSolutions(
+                        handle, dxDesc, wDesc, dyDesc, exhaustiveSearch, false, bufs, invoke_ctx);
+                    const auto fft_solutions = FindFftSolutions(ctx, invoke_ctx);
+                    all_solutions.insert(
+                        all_solutions.end(), direct_solutions.begin(), direct_solutions.end());
+                    all_solutions.insert(all_solutions.end(),
+                                         implictgemm_solutions.begin(),
+                                         implictgemm_solutions.end());
+                    all_solutions.insert(
+                        all_solutions.end(), fft_solutions.begin(), fft_solutions.end());
+                }
             }
 
-            // Implicit GEMM algo
-            if(!use_winograd_only)
+            // Precompile Solutions
             {
-                ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
-                bufs.SetBwd(dx, w, dy);
-                const auto all = this->FindDataImplicitGemmSolutions(
-                    handle, dxDesc, wDesc, dyDesc, exhaustiveSearch, false, bufs, invoke_ctx);
-                PrecompileSolutions(handle, all);
-                const auto algorithm_name =
-                    AlgorithmName{"miopenConvolutionBwdDataAlgoImplicitGEMM"};
-                EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
+                PrecompileSolutions(handle, all_solutions);
             }
 
-            if(!use_winograd_only && !miopen::IsDisabled(MIOPEN_DEBUG_CONV_FFT{}))
+            // Evaluate Invokers
+            auto iter_start = all_solutions.begin();
+            auto iter       = all_solutions.begin();
+            std::string algo_name =
+                solver::Id(iter_start->solver_id).GetAlgo(ctx.conv_problem.GetDirection());
```

Comment:
`conv -x 1 -y 1 -W 7 -H 7 -c 2048 -n 8 -k 512 -p 3 -q 3 -u 2 -v 2 -l 1 -j 1 -g 1 -F 2` segfaults in NORMAL mode with:
<details><summary><b><i>gdb session log</i></b></summary>

```
atamazov@morpheus:~/github/MLOpen2/build/debug.hip-clang$ MIOPEN_FIND_MODE=normal gdb --args ./bin/MIOpenDriver conv -x 1 -y 1 -W 7 -H 7 -c 2048 -n 8 -k 512 -p 3 -q 3 -u 2 -v 2 -l 1 -j 1 -g 1 -F 2 -V 0 -i 1
GNU gdb (Ubuntu 8.1-0ubuntu3) 8.1.0.20180409-git
...
Reading symbols from ./bin/MIOpenDriver...done.
(gdb) run
Starting program: /home/atamazov/github/MLOpen2/build/debug.hip-clang/bin/MIOpenDriver conv -x 1 -y 1 -W 7 -H 7 -c 2048 -n 8 -k 512 -p 3 -q 3 -u 2 -v 2 -l 1 -j 1 -g 1 -F 2 -V 0 -i 1
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
MIOpenDriver conv -x 1 -y 1 -W 7 -H 7 -c 2048 -n 8 -k 512 -p 3 -q 3 -u 2 -v 2 -l 1 -j 1 -g 1 -F 2 -V 0 -i 1
[New Thread 0x7fffd0f87700 (LWP 61602)]
MIOpen(HIP): Info [Handle] stream: 0x1973db0, device_id: 0
MIOpen(HIP): Info [BackwardDataGetWorkSpaceSize]
MIOpen(HIP): Info [HipCompilerVersionImpl] 3.5.20214
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, MIOpen version 2.10.0.8292-95322b87a
MIOpen(HIP): Info [GetFindModeValueImpl] MIOPEN_FIND_MODE = NORMAL(1)
[New Thread 0x7ffec53ff700 (LWP 61605)]
[Thread 0x7ffec53ff700 (LWP 61605) exited]
[New Thread 0x7ffff7f62700 (LWP 61606)]
MIOpen(HIP): Info [FindConvBwdDataAlgorithm] requestAlgoCount = 2, workspace = 401408
MIOpen(HIP): Info [TryLoad] Find-db regenerating.

Thread 1 "MIOpenDriver" received signal SIGSEGV, Segmentation fault.
0x00007fffd2665470 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::c_str() const () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
(gdb) bt
#0  0x00007fffd2665470 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::c_str() const () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#1  0x00007ffff3509614 in miopen::solver::Id::Id (this=0x7fffffff9368, Python Exception <class 'gdb.error'> There is no member named _M_dataplus.:str=) at /home/atamazov/github/MLOpen2/src/solver.cpp:133
#2  0x00007ffff397eca4 in miopen::ConvolutionDescriptor::FindConvBwdDataAlgorithm(miopen::Handle&, miopen::TensorDescriptor const&, void const*, miopen::TensorDescriptor const&, void const*, miopen::TensorDescriptor const&, void*, int, int*, miopenConvAlgoPerf_t*, void*, unsigned long, bool) const::{lambda(miopen::DbRecord&)#1}::operator()(miopen::DbRecord&) const (this=0x19e7e60, record=...) at /home/atamazov/github/MLOpen2/src/ocl/convolutionocl.cpp:2105
#3  0x00007ffff397e1d2 in std::_Function_handler<void (miopen::DbRecord&), miopen::ConvolutionDescriptor::FindConvBwdDataAlgorithm(miopen::Handle&, miopen::TensorDescriptor const&, void const*, miopen::TensorDescriptor const&, void const*, miopen::TensorDescriptor const&, void*, int, int*, miopenConvAlgoPerf_t*, void*, unsigned long, bool) const::{lambda(miopen::DbRecord&)#1}>::_M_invoke(std::_Any_data const&, miopen::DbRecord&) (__functor=..., __args=...) at /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/std_function.h:316
#4  0x00007ffff3993d0b in std::function<void (miopen::DbRecord&)>::operator()(miopen::DbRecord&) const (this=0x7fffffffa660, __args=...) at /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/std_function.h:706
#5  0x00007ffff398fd38 in miopen::FindDbRecord_t<miopen::PlainTextDb>::TryLoad<miopen::ProblemDescription>(miopen::Handle&, miopen::ProblemDescription const&, std::function<void (miopen::DbRecord&)> const&) (handle=..., problem=..., regenerator=...) at /home/atamazov/github/MLOpen2/src/include/miopen/find_db.hpp:154
#6  0x00007ffff395f599 in miopen::ConvolutionDescriptor::FindConvBwdDataAlgorithm (this=0x19712e0, handle=..., dyDesc=..., dy=0x7ffec4809000, wDesc=..., w=0x7ffebd200000, dxDesc=..., dx=0x7ffec5400000, requestAlgoCount=2, returnedAlgoCount=0x7fffffffd90c, perfResults=0x19eca10, workSpace=0x7ffec48cd000, workSpaceSize=401408, exhaustiveSearch=false) at /home/atamazov/github/MLOpen2/src/ocl/convolutionocl.cpp:2055
#7  0x00007ffff32fc87c in miopenFindConvolutionBackwardDataAlgorithm::{lambda()#2}::operator()() const (this=0x7fffffffb380) at /home/atamazov/github/MLOpen2/src/convolution_api.cpp:1021
#8  0x00007ffff32ee9ec in miopen::try_<miopenFindConvolutionBackwardDataAlgorithm::{lambda()#2}>(miopenFindConvolutionBackwardDataAlgorithm::{lambda()#2}, bool) (f=..., output=true) at /home/atamazov/github/MLOpen2/src/include/miopen/errors.hpp:76
#9  0x00007ffff32ee7d0 in miopenFindConvolutionBackwardDataAlgorithm (handle=0x19ac700, dyDesc=0x1973920, dy=0x7ffec4809000, wDesc=0x1971ff0, w=0x7ffebd200000, convDesc=0x19712e0, dxDesc=0x1975380, dx=0x7ffec5400000, requestAlgoCount=2, returnedAlgoCount=0x7fffffffd90c, perfResults=0x19eca10, workSpace=0x7ffec48cd000, workSpaceSize=401408, exhaustiveSearch=false) at /home/atamazov/github/MLOpen2/src/convolution_api.cpp:1020
#10 0x0000000000499056 in ConvDriver<float, float>::FindBackwardData (this=0x18e86f0, ret_algo_count=@0x7fffffffd90c: 0, request_algo_count=2, perf_results=std::vector of length 2, capacity 2 = {...}, ctx=0) at /home/atamazov/github/MLOpen2/driver/conv_driver.hpp:1960
#11 0x0000000000495f06 in ConvDriver<float, float>::RunBackwardDataGpuFind (this=0x18e86f0) at /home/atamazov/github/MLOpen2/driver/conv_driver.hpp:2085
#12 0x00000000004328e7 in ConvDriver<float, float>::RunBackwardGPU (this=0x18e86f0) at /home/atamazov/github/MLOpen2/driver/conv_driver.hpp:2022
#13 0x00000000004129cf in main (argc=36, argv=0x7fffffffe028) at /home/atamazov/github/MLOpen2/driver/main.cpp:218
```
</details>

Reply:
This is acctually a bug. I've fixed it.